Update SerializationUtilTest.java

agrawalprakash opened the pull request:
Added additional test to improve test coverage of SerializationUtil.java

devOpsHazelcast commented:
Can one of the admins verify this patch?

devOpsHazelcast commented:
Can one of the admins verify this patch?

devOpsHazelcast commented:
Can one of the admins verify this patch?

devOpsHazelcast commented:
Can one of the admins verify this patch?

devOpsHazelcast commented:
Can one of the admins verify this patch?

devOpsHazelcast commented:
Can one of the admins verify this patch?

devOpsHazelcast commented:
Can one of the admins verify this patch?

devOpsHazelcast commented:
Can one of the admins verify this patch?

devOpsHazelcast commented:
Can one of the admins verify this patch?

devOpsHazelcast commented:
Can one of the admins verify this patch?

devOpsHazelcast commented:
Can one of the admins verify this patch?

devOpsHazelcast commented:
Can one of the admins verify this patch?

devOpsHazelcast commented:
Can one of the admins verify this patch?

devOpsHazelcast commented:
Can one of the admins verify this patch?

devOpsHazelcast commented:
Can one of the admins verify this patch?

devOpsHazelcast commented:
Can one of the admins verify this patch?

devOpsHazelcast commented:
Can one of the admins verify this patch?

devOpsHazelcast commented:
Can one of the admins verify this patch?

devOpsHazelcast commented:
[![CLA assistant check](https://cla.hazelcast.org/pull/badge/not_signed)](https://cla.hazelcast.org/hazelcast/hazelcast?pullRequest=23757) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.hazelcast.org/hazelcast/hazelcast?pullRequest=23757) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.hazelcast.org/check/hazelcast/hazelcast?pullRequest=23757) it.</sub>

pveentjer commented:
@ihsandemir @mdumandag could you have a look?

mdumandag commented on hazelcast/src/test/java/com/hazelcast/internal/serialization/impl/SerializationUtilTest.java:
```diff
@@ -102,6 +108,21 @@ public void testReadWriteNullableBoolean_whenInvalid()
+
+        serializationUtil.getInterfaces(type, interfaces);
+
+        verify(serializationUtil, times(1)).getInterfaces(type, interfaces);
```
The test currently does not verify anything other than the call to `serializationUtil.getInterfaces(type, interfaces);` is made(which we call in the test method). I don't think it is useful in the current state for anything other than increasing the coverage

agrawalprakash replied:
Thank you, agree this was for coverage. If coverage of the method reduces in future then it will indicate to enhance the test or write a new one.

mdumandag replied:
I don't like writing a test just to artificially increase the coverage. We should assert that the method does its job, collects the interfaces to the passed collection

agrawalprakash replied:
Hi, I have removed this test which I will check later and I have added test for method SerializationUtil.createSerializerAdapter where I am able to assert the results. Please can you review if it looks good.

mdumandag commented on hazelcast/src/test/java/com/hazelcast/internal/serialization/impl/SerializationUtilTest.java:
```diff
@@ -102,6 +108,21 @@ public void testReadWriteNullableBoolean_whenInvalid()
+        SerializationUtil serializationUtil = mock(SerializationUtil.class);
+
+       // HashSet implements Set<E>, Cloneable, java.io.Serializable, hence using it for this test
+        Class type = new HashSet().getClass();
```
```suggestion
        Class type = HashSet.class;
```

agrawalprakash replied:
Thank you for the suggestion, I have updated my pull request with a new commit to fix this.

mdumandag commented on hazelcast/src/test/java/com/hazelcast/internal/serialization/impl/SerializationUtilTest.java:
```diff
@@ -102,6 +108,21 @@ public void testReadWriteNullableBoolean_whenInvalid()
 
+    @Test
+    public void testGetInterfacesCalledVerified() {
+
```
minor: unnecessary newline

agrawalprakash replied:
Thank you for the suggestion, I have updated my pull request with a new commit to fix this.

mdumandag commented on hazelcast/src/test/java/com/hazelcast/internal/serialization/impl/SerializationUtilTest.java:
```diff
@@ -102,6 +108,21 @@ public void testReadWriteNullableBoolean_whenInvalid()
+        serializationUtil.getInterfaces(type, interfaces);
+
+        verify(serializationUtil, times(1)).getInterfaces(type, interfaces);
+
```
minor: unnecessary new line

agrawalprakash replied:
Thank you for the suggestion, I have updated my pull request with a new commit to fix this.

agrawalprakash commented:
@mdumandag  Hi, Please can you review and approve if this commit looks ok now

agrawalprakash commented:
@mdumandag  Hi, I have removed the earlier test and added test for method SerializationUtil.createSerializerAdapter where I assert the results. Please can you review if it looks good.

mdumandag commented on hazelcast/src/test/java/com/hazelcast/internal/serialization/impl/SerializationUtilTest.java:
```diff
@@ -102,6 +109,17 @@ public void testReadWriteNullableBoolean_whenInvalid()
 
+    @Test
+    public void testCreateSerializerAdapter() {
+        ClassNameFilter classNameFilter = new SerializationClassNameFilter(new JavaSerializationFilterConfig());
+        // JavaSerializer is instance of StreamSerializer, hence using it as parameter
+        SerializerAdapter streamSerializerAdapter = SerializationUtil.createSerializerAdapter(new JavaSerializer(true, true, classNameFilter));
```
```suggestion
        // ArrayStreamSerializer is instance of StreamSerializer, hence using it as parameter
        SerializerAdapter streamSerializerAdapter = SerializationUtil.createSerializerAdapter(new ArrayStreamSerializer());
```
use this instead to simplify the test

agrawalprakash replied:
@mdumandag  Hi, I have made the changes for this and updated this pull request.

mdumandag commented on hazelcast/src/test/java/com/hazelcast/internal/serialization/impl/SerializationUtilTest.java:
```diff
@@ -16,31 +16,38 @@
-import static org.junit.Assert.assertNull;
-import static org.junit.Assert.assertTrue;
-import static org.mockito.Mockito.mock;
+import com.hazelcast.test.annotation.QuickTest;;
```
```suggestion
import com.hazelcast.test.annotation.QuickTest;
```

agrawalprakash replied:
@mdumandag  Hi, I have made the changes for this and updated this pull request.

mdumandag commented:
run-lab-run
